src/internal/connector/graph/collections.go

```diff
@@ -48,22 +48,28 @@ func (c emptyCollection) DoNotMergeItems() bool {
 
 func BaseCollections(
 	ctx context.Context,
+	colls []data.BackupCollection,
 	tenant, rOwner string,
 	service path.ServiceType,
 	categories map[path.CategoryType]struct{},
 	su support.StatusUpdater,
 	errs *fault.Bus,
 ) ([]data.BackupCollection, error) {
 	var (
-		res     = []data.BackupCollection{}
-		el      = errs.Local()
-		lastErr error
+		res      = []data.BackupCollection{}
+		el       = errs.Local()
+		lastErr  error
+		collKeys = map[string]struct{}{}
 	)
 
+	for _, c := range colls {
+		collKeys[c.FullPath().String()] = struct{}{}
+	}
```

Comment:
hmmm looking at this with fresh eyes this needs a comment and a fixup

fixup: it's not going to work for collections denoting deleted folders because they have a nil `FullPath` so I think it will NPE

comment: this will not guarantee getting all collections, only those that aren't deleted (see above). It may be less confusing to only track collections related to the prefix directories?